diff -up xdg-desktop-portal-1.20.3/src/xdp-portal-impl.c.1~ xdg-desktop-portal-1.20.3/src/xdp-portal-impl.c
--- xdg-desktop-portal-1.20.3/src/xdp-portal-impl.c.1~	2025-05-20 19:07:14.000000000 +0200
+++ xdg-desktop-portal-1.20.3/src/xdp-portal-impl.c	2025-06-18 01:29:32.995833856 +0200
@@ -807,6 +807,40 @@ add_all_default_portal_implementations_i
 static XdpPortalImplementation *
 find_gtk_fallback_portal_implementation (const char *interface)
 {
+  /* Let's try returning something nice first, gtk file dialogs stink */
+  for (size_t i = 0; i < implementations->len; i++)
+    {
+      XdpPortalImplementation *impl = g_ptr_array_index (implementations, i);
+
+      if (!g_str_equal (impl->dbus_name, "org.freedesktop.impl.portal.desktop.kde"))
+        continue;
+
+      if (!portal_impl_supports_iface (impl, interface))
+        continue;
+
+      g_warning ("Choosing %s.portal for %s as a last-resort fallback",
+                 impl->source, interface);
+
+      return impl;
+    }
+
+  for (size_t i = 0; i < implementations->len; i++)
+    {
+      XdpPortalImplementation *impl = g_ptr_array_index (implementations, i);
+
+      if (!g_str_equal (impl->dbus_name, "org.freedesktop.impl.portal.desktop.lxqt"))
+        continue;
+
+      if (!portal_impl_supports_iface (impl, interface))
+        continue;
+
+      g_warning ("Choosing %s.portal for %s as a last-resort fallback",
+                 impl->source, interface);
+
+      return impl;
+    }
+
+
   /* As a last resort, if nothing was selected for this desktop by
    * ${desktop}-portals.conf or portals.conf, and no portal volunteered
    * itself as suitable for this desktop via the legacy UseIn mechanism,
